Sources/Lithosphere/DiagnosticConsumer.swift

```diff
@@ -10,7 +10,7 @@ import Foundation
 /// diagnostics into its internal storage. It's responsible for converting
 /// the content of the diagnostics into a different form for serialization,
 /// textual output, and whatever else it wants.
-public protocol DiagnosticConsumer {
+public protocol DiagnosticConsumer: AnyObject {
```

Comment:
Would it be overkill to drop this and hand back an `NSNotificationCenter`-style token that has to be used for de-registering consumers?

Reply:
I suppose I could do that, yeah!